### DISCUSSION TITLE
Unable to deploy the function

### DISCUSSION THREAD

**Discussion by octonawish-akcodes** [knative/func#2405] [Q&A]
I am unable to deploy the knative function using func command, the build image is successful and locally tested too but when it comes to deploy it even after providing correct credentials to my docker HUB its still fails, I am stuck what to do or I'm missing anything here?
Can I get some guidance here?
![image](https://github.com/knative/func/assets/76171953/47079049-e53a-4deb-bdde-0cb465c4c9e7)

**Comment by lkingland** [comment]
It looks like you're having trouble logging in.  Perhaps try running `docker login` first.  The credentials you enter should be used by the deployment step automatically.
